mypy/semanal_pass3.py

```diff
@@ -471,12 +472,18 @@ def patch2() -> None:
 
             self.patches.append((PRIORITY_TYPEVAR_VALUES, self.make_scoped_patch(patch2)))
 
-    def analyze_info(self, info: TypeInfo) -> None:
+    def analyze_synthetic_info(self, info: TypeInfo) -> None:
         # Similar to above but for nodes with synthetic TypeInfos (NamedTuple and NewType).
         for name in info.names:
             sym = info.names[name]
             if isinstance(sym.node, (FuncDef, Decorator)):
+                # Since we are analyzing a synthetic type info, the methods there
+                # are not real independent targets, and should be processed when
+                # the enclosing synthetic type is processed.
+                old_recurse = self.recurse_into_functions
+                self.recurse_into_functions = True
```

Comment:
If this is not needed to fix the test cases, could you move this to a separate PR?

Reply:
Hm, I think I can split it by moving one of the tests added to the second PR.